### PR TITLE
Sourcery Starbot ⭐ refactored Lucas8x/luscious-downloader

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>main</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/luscious-downloader main
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: luscious_dl/utils.py [hunk 1]
```diff
@@ -47,14 +47,11 @@ def format_foldername(album, foldername_format: str = '%t') -> str:
...
-      .replace('%a', album.author) \
-      .replace('%p', str(album.number_of_pictures)) \
-      .replace('%g', str(album.number_of_animated_pictures)) \
-      .replace('[]', '').strip()
-  return folder_name
+  return (foldername_format.replace('%i', str(album.id_)).replace(
+      '%t', album_name).replace('%a', album.author).replace(
+          '%p', str(album.number_of_pictures)).replace(
+              '%g', str(album.number_of_animated_pictures)).replace(
+                  '[]', '').strip())
```
Comment (SourceryAI) [inline luscious_dl/utils.py:54]: Function `format_foldername` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/refactorings/inline-immediately-returned-variable/))

File: luscious_dl/utils.py [hunk 2]
```diff
@@ -104,14 +101,14 @@ def generate_pdf(output_dir: Path, formmatted_name: str, album_folder: Path) ->
...
-    if len(pictures_path_list) > 0:
+    if pictures_path_list:
       for picture_path in tqdm(pictures_path_list, desc='Opening/Coverting images'):
         img = Image.open(picture_path)
         if picture_path.suffix.lower() == '.png' or img.mode == 'RGBA':
           img = img.convert('RGB')
         pictures.append(img)
 
-    if len(pictures) == 0:
+    if not pictures:
```
Comment (SourceryAI) [inline luscious_dl/utils.py:111]: Function `generate_pdf` refactored with the following changes:

- Simplify sequence length comparison [×2] ([`simplify-len-comparison`](https://docs.sourcery.ai/refactorings/simplify-len-comparison/))

File: luscious_dl/utils.py [hunk 3]
```diff
@@ -163,7 +160,7 @@ def read_list(directory: Path) -> list[str]:
       logger.log(5, f'Total of Items: {len(list_txt)}.')
     return list(set(list_txt))
   except FileNotFoundError:
-    logger.error(f"The list.txt file doesn't exist in this folder.")
+    logger.error("The list.txt file doesn't exist in this folder.")
```
Comment (SourceryAI) [inline luscious_dl/utils.py:163]: Function `read_list` refactored with the following changes:

- Replace f-string with no interpolated values with string ([`remove-redundant-fstring`](https://docs.sourcery.ai/refactorings/remove-redundant-fstring/))

File: luscious_dl/utils.py [hunk 4]
```diff
@@ -187,8 +184,7 @@ def get_config_data() -> dict:
   """
   try:
     with get_root_path().joinpath('config.json').open() as config:
-      data = json.load(config)
-      return data
+      return json.load(config)
```
Comment (SourceryAI) [inline luscious_dl/utils.py:187]: Function `get_config_data` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/refactorings/inline-immediately-returned-variable/))

File: luscious_dl/utils.py [hunk 5]
```diff
@@ -220,21 +216,29 @@ def create_default_files() -> None:
...
+      user_inputs=None,
+      read_list=False,
+      only_favorites=False,
+      keyword=None,
+      search_download=False,
+      sorting='date_trending',
+      page=1,
+      max_pages=1,
   )
-  return base_namespace
```
Comment (SourceryAI) [inline luscious_dl/utils.py:237]: Function `load_settings` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/refactorings/inline-immediately-returned-variable/))

File: luscious_dl/utils.py [hunk 6]
```diff
@@ -344,10 +348,13 @@ def open_config_menu() -> None:
...
-          if input('\nNo album identifiers found.\nYou sure? ("Y/N")\n> ') in 'nN':
-            print('\nAlbum folder format set to the default.\n')
-            new_folderformat = '%t'
+        if (all(
+            identifier not in new_folderformat
+            for identifier in ('%i', '%t', '%a'))
+            and input('\nNo album identifiers found.\nYou sure? ("Y/N")\n> ')
+            in 'nN'):
+          print('\nAlbum folder format set to the default.\n')
+          new_folderformat = '%t'
```
Comment (SourceryAI) [inline luscious_dl/utils.py:357]: Function `open_config_menu` refactored with the following changes:

- Merge nested if conditions ([`merge-nested-ifs`](https://docs.sourcery.ai/refactorings/merge-nested-ifs/))
- Invert any/all to simplify comparisons ([`invert-any-all`](https://docs.sourcery.ai/refactorings/invert-any-all/))

**Comment by Rage2025** [comment]
The Luscious Downloader is discontinued forever because of the external request

**Pull request closed without merging**